keras_nlp/transformer/transformer_decoder.py

```diff
@@ -0,0 +1,193 @@
+import tensorflow as tf
+from keras import layers
+
+
+class TransformerDecoder(keras.layers.Layer):
+    """Transformer decoder.
+
+    This class follows the architecture of transformer decoder layer in paper
+    "Attention is All You Need" (https://arxiv.org/abs/1706.03762). Users can
+    instantiate multiple instances of this class to stack up the decoder.
+
+    Args:
+        feedforward_dim: int, the hidden size of feedforward network.
+        num_heads: int, the number of heads in MultiHeadAttention.
+        dropout: float, defaults to 0. the dropout value, shared by
+            MultiHeadAttention and feedforward network.
+        activation: string or tf.keras.activations, defaults to "relu". the
+            activation function of feedforward network.
+        layer_norm_epsilon: float, defaults to 1e-5. The eps value in layer
+            normalization components.
+        name: string, defaults to "TransformerDecoder". The name of the layer.
+        **kwargs: other keyword arguments.
+
+    Examples:
+
+    ```python
+    # Create a single transformer decoder layer.
```

Comment:
We should figure out if we want the testable docstrings a la core keras. We could also submit as is and add later.